nbgrader/api.py

```diff
@@ -443,6 +322,191 @@ def __repr__(self):
         return "{} for {}".format(self.cell, self.student)
 
 
+## Needs manual grade
+
+SubmittedNotebook.needs_manual_grade = column_property(
+    exists().where(and_(
+        Grade.notebook_id == SubmittedNotebook.id,
+        Grade.needs_manual_grade))\
+    .correlate_except(Grade), deferred=True)
+
+SubmittedAssignment.needs_manual_grade = column_property(
+    exists().where(and_(
+        SubmittedNotebook.assignment_id == SubmittedAssignment.id,
+        Grade.notebook_id == SubmittedNotebook.id,
+        Grade.needs_manual_grade))\
+    .correlate_except(Grade), deferred=True)
+
+
+## Overall scores
+
+SubmittedNotebook.score = column_property(
+    select([func.coalesce(func.sum(Grade.score), 0.0)])\
+        .where(Grade.notebook_id == SubmittedNotebook.id)\
+        .correlate_except(Grade), deferred=True)
+
+SubmittedAssignment.score = column_property(
+    select([func.coalesce(func.sum(Grade.score), 0.0)])\
+        .where(and_(
+            SubmittedNotebook.assignment_id == SubmittedAssignment.id,
+            Grade.notebook_id == SubmittedNotebook.id))\
+        .correlate_except(Grade), deferred=True)
+
+Student.score = column_property(
+    select([func.coalesce(func.sum(Grade.score), 0.0)])\
+        .where(and_(
+            SubmittedAssignment.student_id == Student.id,
+            SubmittedNotebook.assignment_id == SubmittedAssignment.id,
+            Grade.notebook_id == SubmittedNotebook.id))\
+        .correlate_except(Grade), deferred=True)
+
```

Comment:
My SqlAlchemy Knowleged is ~0, but the 3 above column seem to be "just" one more `where` clause each time. 
Would there be a way to simplify ? (No, or not worth it is a good response)